nemo/collections/cv/models/mnist_lenet5.py

```diff
@@ -0,0 +1,146 @@
+    module: Config = Config()
+
+
+@experimental
+class MNISTLeNet5(ModelPT):
+    def __init__(self, cfg: MNISTLeNet5Config = MNISTLeNet5Config()):
+        super().__init__()
+        # Remember config - should be moved to base init.
+        self._cfg = cfg
+
+        # Initialize modules.
+        self.module = LeNet5Module(cfg.module)
+        self.loss = NLLLoss()
+
+    @property
+    def input_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.module.input_types
+
+    @property
+    def output_types(self) -> Optional[Dict[str, NeuralType]]:
+        return self.module.output_types
+
+    @typecheck()
+    def forward(self, images):
+        return self.module.forward(images=images)
+
+    def configure_optimizers(self):
+        # Get optimizer class.
+        optimizer_cls = get_optimizer(self._cfg.opt.cls)
+        # Instantiate the optimizer, set model parameters and pass other kwargs.
+        optimizer = optimizer_cls(params=self.parameters(), **self._cfg.opt.params)
+        return optimizer
+
+    def training_step(self, batch, what_is_this_input):
+        # "Unpack" the batch.
+        _, images, targets, _ = batch
+
+        # Get predictions.
+        predictions = self(images=images)
+
+        # Calculate loss.
+        loss = self.loss(predictions=predictions, targets=targets)
+
+        # Return it.
+        return {"loss": loss}
+        # of course "return loss" doesn't work :]
+
+    def train_dataloader(self):
+        """ Create dataset, wrap it with dataloader and return the latter """
+        # Instantiate dataset.
+        mnist_ds = MNISTDataset(self._cfg.dataset)
+        # Configure data loader.
+        train_dataloader = DataLoader(dataset=mnist_ds, **(self._cfg.dataloader))
+
+        return train_dataloader
+
+    def setup_training_data(self, train_data_layer_params: Optional[Dict]):
+        """ Dummy methods """
+        pass
+
+    def setup_validation_data(self, val_data_layer_params: Optional[Dict]):
+        """ Dummy methods """
+        pass
+
+    def setup_test_data(self, test_data_layer_params: Optional[Dict]):
+        """ Dummy methods """
+        pass
+
+    def setup_optimization(self, optim_params: Optional[Dict] = None) -> Optimizer:
+        """ Dummy methods """
+        pass
+
+    def save_to(self, save_path: str):
+        """ Why do I need that in experimental module? """
+        pass
+
+    @classmethod
+    def restore_from(cls, restore_path: str):
+        """ Why do I need that in experimental module? """
+        pass
+
+    @classmethod
+    def list_available_models(cls) -> Optional[Dict[str, str]]:
+        """ Why do I need that in experimental module? """
+        pass
+
+    @classmethod
+    def from_pretrained(cls, name: str):
+        """ Why do I need that in experimental module? """
+        pass
+
+    def export(self, **kwargs):
+        """ Why do I need that in experimental module? """
+        pass
```

Comment:
Please remove all unnecessary docstrings. I count 9 of them

Reply:
Done.